tests/test_api.rs

```diff
@@ -9566,6 +9566,64 @@ fn test_fast_calls() {
   assert_eq!("fast", unsafe { WHO });
 }
 
+#[test]
+fn test_fast_calls_empty_buffer() {
+  static mut WHO: &str = "none";
+  unsafe fn fast_fn(
+    _recv: v8::Local<v8::Object>,
+    buffer: *mut FastApiTypedArray<u8>,
+  ) {
+    assert_eq!(WHO, "slow");
+    WHO = "fast";
+    assert_eq!(
+      0,
+      FastApiTypedArray::get_storage_from_pointer_if_aligned(buffer)
+        .unwrap()
+        .len()
+    );
+  }
+
+  fn slow_fn(
+    _scope: &mut v8::HandleScope,
+    _args: v8::FunctionCallbackArguments,
+    _rv: v8::ReturnValue,
+  ) {
+    unsafe {
+      WHO = "slow";
+    }
+  }
+
+  const FAST_TEST: fast_api::FastFunction = fast_api::FastFunction::new(
+    &[V8Value, TypedArray(CType::Uint8)],
+    fast_api::CType::Void,
+    fast_fn as _,
+  );
+
+  let _setup_guard = setup::parallel_test();
+  let isolate = &mut v8::Isolate::new(Default::default());
+  let scope = &mut v8::HandleScope::new(isolate);
+  let context = v8::Context::new(scope);
+  let scope = &mut v8::ContextScope::new(scope, context);
+
+  let global = context.global(scope);
+
+  let template = v8::FunctionTemplate::builder(slow_fn)
+    .build_fast(scope, &FAST_TEST, None, None, None);
+
+  let name = v8::String::new(scope, "func").unwrap();
+  let value = template.get_function(scope).unwrap();
+  global.set(scope, name.into(), value.into()).unwrap();
+  let source = r#"
+    function f(arr) { func(arr); }
+    %PrepareFunctionForOptimization(f);
+    f(new Uint8Array(0));
```

Comment:
**note:** This sort of `f(new Uint8Array())` is precisely the case where the V8 bug of an invalid backing pointer may happen. It's precisely when it's a fast call being done in this fashion.

I've actually documented this in somewhat explicit detail in the most logical place of them all: Deno FFI tests https://github.com/denoland/deno/blob/main/test_ffi/tests/test.js#L369-L373 :D